File: server/src/main/java/com/abdelbahmadi/service/ProfileService.java
package com.abdelbahmadi.service;

import com.abdelbahmadi.models.Profile;
import com.abdelbahmadi.response.*;
import org.springframework.web.bind.annotation.PathVariable;

import java.util.List;

public interface ProfileService {
    ProfileDTO editProfile(ProfileDTO profileDTO,  Integer profileId, Integer userId);
    ProfileDTO getProfile(Integer userId);
    ProfileSelectionDTO getProfileSelection(String bearerToken);
    List<Profile> searchUser(String query);
}

File: client/src/service/profile.service.js
import instance from "../api/apiConfig";
import { PROFILE_URL } from "../utils";

export const profileService = {
  getProfile: async (userId) => {
    try {
      const response = await instance.get(`${PROFILE_URL}/${userId}`);
      return response.data;
    } catch (error) {
      console.log(error);
    }
  },
  editProfile: async ({profileDto, profileId, currentUserId}) => {
    try {
      const response = await instance.put(`${PROFILE_URL}/${profileId}/${currentUserId}`, profileDto);
      return response.data;
    } catch (error) {
      console.log(error);
    }
  }
};
